### ISSUE TITLE
Callback doesn't fire

### ISSUE THREAD

**Issue by webdes03** [DaveGut/TP-Link-to-SmartThings-Integration#1]
I've setup your TP-LinkServer on a Raspberry Pi running Node. I'm able to issue commands to it just fine through tools like Postman by setting the following headers:

Method: Get
Url: http://nnn.nnn.nnn.nnn:8082
Headers: tplink-iot-ip - nnn.nnn.nnn.nnn, and command - {"command":"{\"system\":{\"set_relay_state\":{\"state\": 1}}}"}

This works perfectly, the server shows the request in the output, the outlet (a HS-100) turns on/off as expected based on the state in the headers.

However, when I set it up through SmartThings, it doesn't work. Status changes to "Turning On", but never actually does anything. I tried adding some additional logging into TP-LInk_Plug_Switch_v2.0.groovy, and have diagnosed it down to the sendHubCommand call inside sendCmdtoServer, not firing, or at least the callback never fires, so I'm not sure if the hub command is failing, or there's some other gremlin in there.

The device in question is configured to use the TP-Link_Plug_Switch type, and has the correct deviceIP and gatewayIP that were used in the previous Postman test, so I'm really struggling to figure out if it's a gremlin in my environment or some other sort of issue.

SmartThings hub ison Firmware 000.016.00013

Any thoughts?

**Comment by DaveGut** [comment]
Sorry for the delay.  Very sorry.  I do not monitor here enough. Will start more.
 
My version info
Hub version 12 (essentially same as yours)
Node.js version - original - 6.9.4 (used for development)
Just updated to node.js version 6.10.3  -  CAUSED PROBLEM

There is a problem (today) in newer node.js versions.  For some reason, the socket.write (in "TP-LinkServer.js") is not writing.  I loaded the newer version of node.js (6.10.3, a fresh install).  It did not work.  Then, setting up traces through console.log statements, I inserted one after three lines (at the arrows below) to see where they stopped.

function processDeviceCommand(request, response) {
	var command = request.headers["tplink-command"]
	var deviceIP = request.headers["tplink-iot-ip"]
	console.log("Sending to IP address: " + deviceIP + " Command: " + command)
//--- Encrypt then send command to device and wait for response. -------
	return new Promise((resolve, reject) => {
		var socket = net.connect(9999, deviceIP)
		socket.setKeepAlive(false)
		socket.setTimeout(2000)  // 2 seconds.
   	 	socket.on('connect', () => {
--->> insert console.log here
  			socket.write(encrypt(command))
--->> insert console.log here
   	 	})
//-- Decrypt response (less header) then send to SmartThings. ------------------
--->> insert console.log here
		socket.on('data', (data) => {

Started the applet and tried command.  It did not work.  Tried again (no other changes), it worked!!!

I reloaded the original file and it worked perfectly the first time.

I uninstalled and reinstalled node.js 6.10.3 to try to duplicate initial conditions.  However it work at first run.  Try the above or try rebooting your raspberry.  I do not know why this is occurring.  Any ideas?
